### ISSUE TITLE
If an error happens, the original file is reduced to zero bytes!

### ISSUE THREAD

**Issue by antoinerousseau** [stefansl/image-shrinker#54]
I'm trying to shrink [this 2.5M image](https://user-images.githubusercontent.com/160279/64801432-ce01be00-d588-11e9-99a0-b882f2eb35db.jpg), and I get the error below, but at the same time my file is reduced to zero bytes, although the "Add .min sufffix" is enabled!

<img width="532" alt="Capture d'écran 2019-09-12 18 11 30" src="https://user-images.githubusercontent.com/160279/64801565-0f926900-d589-11e9-870e-642d3c47c71d.png">

Image Shrinker version: 1.5.0
macOS version: 10.14.6

**Comment by stefansl** [comment]
It must be an error in execFile here: https://github.com/stefansl/image-shrinker/blob/66fcf0b1f5a89f96ddda863b0cf4457fa99ef9f0/main.js#L246
If it thows an error (which it does in your case) the error will be logged in your native log of macOS (check your Console.app). Can you find any entries here?

**Comment by antoinerousseau** [comment]
The error in the Console is:
```
nw_path_close_fd Failed to close guarded necp fd 31 [9: Bad file descriptor]
```

**Comment by antoinerousseau** [comment]
I just tried to use mozjpeg's `cjpeg` bin in the commandline and it worked, but only if the `-outfile` is different than the input file, e.g. `cjpeg -outfile test.jpg test.jpg` fails with a zero-byte file.

Also, have you considered using [imagemin-mozjpeg](https://github.com/imagemin/imagemin-mozjpeg) as mentioned in [the README of the lib you use](https://github.com/imagemin/mozjpeg-bin#mozjpeg-bin-)?

**Comment by stefansl** [comment]
Yes, I did. But the problem with imagemin is, you can't control the output. Means, I didn't find any possibility to change the filename to .min. But in the end imagemin is just a wrapper ... 

The problem is now, it's not reproducable. The only thing I could do is to create a tmp copy from the original image, so that this will not be destroyed in case of an error.

**Comment by antoinerousseau** [comment]
But since the output should be another file ending with .min.jpg, why is it writing on the original file anyway? (the .min option is on in my settings)

**Comment by stefansl** [comment]
That's should not happen, of course. Is it a fresh install or did you update?
Maybe you can try the develop branch: https://github.com/stefansl/image-shrinker/tree/develop

**Comment by antoinerousseau** [comment]
Just tried it with the develop branch:

<img width="488" alt="Capture d'écran 2019-09-16 11 08 21" src="https://user-images.githubusercontent.com/160279/64946086-5b3b6000-d872-11e9-8754-6e8c606dd657.png">

**Comment by stefansl** [comment]
Thanks for your help, @antoinerousseau !
So now it makes sense according to this issue: https://github.com/mozilla/mozjpeg/issues/248
The only thing I ask myself, why the Shrinker does ignore the "add .min" setting. Please try to change the settings and then change it back. Maybe it works then?

**Comment by antoinerousseau** [comment]
Indeed, switching the setting off and on again worked!

**Comment by stefansl** [comment]
Ok, I will check the initial settings. Thx for your help!

**Comment by stefansl** [comment]
Fixed in 09e5d89210c2b4c9d5c0c0735c1f79a4ed42384b

**Comment by despecial** [comment]
Can you please release an update?
version 1.5.0 still empties jpg files

Thanks
